rllib/utils/test_utils.py

```diff
@@ -28,34 +29,47 @@
 
 logger = logging.getLogger(__name__)
 
+# Dict holding the run-times of each frameworks' `framework_iterator`.
+# This only gets written to if `framework_iterator()` is called with the
+# `time_iterations=True` argument.
+FRAMEWORK_ITERATOR_LAST_RUN_TIMES = {}
 
-def framework_iterator(config=None,
-                       frameworks=("tf2", "tf", "tfe", "torch"),
-                       session=False,
-                       with_eager_tracing=False):
+
+def framework_iterator(
+        config: Optional[PartialTrainerConfigDict] = None,
+        frameworks: Sequence[str] = ("tf2", "tf", "tfe", "torch"),
+        session: bool = False,
+        with_eager_tracing: bool = False,
+        time_iterations: bool = False,
+) -> Union[str, Tuple[str, Optional["tf1.Session"]]]:
     """An generator that allows for looping through n frameworks for testing.
 
     Provides the correct config entries ("framework") as well
     as the correct eager/non-eager contexts for tfe/tf.
 
     Args:
-        config (Optional[dict]): An optional config dict to alter in place
-            depending on the iteration.
-        frameworks (Tuple[str]): A list/tuple of the frameworks to be tested.
+        config: An optional config dict to alter in place depending on the
+            iteration.
+        frameworks: A list/tuple of the frameworks to be tested.
             Allowed are: "tf2", "tf", "tfe", "torch", and None.
-        session (bool): If True and only in the tf-case: Enter a tf.Session()
+        session: If True and only in the tf-case: Enter a tf.Session()
             and yield that as second return value (otherwise yield (fw, None)).
             Also sets a seed (42) on the session to make the test
             deterministic.
         with_eager_tracing: Include `eager_tracing=True` in the returned
             configs, when framework=[tfe|tf2].
+        time_iterations: Whether to time each (framework's) iteration.
+            The timing results can be retrieved after the generator finished
+            via `from ray.rllib.utils.test_utils import
+            FRAMEWORK_ITERATOR_LAST_RUN_TIMES`.
 
     Yields:
-        str: If enter_session is False:
-            The current framework ("tf2", "tf", "tfe", "torch") used.
-        Tuple(str, Union[None,tf.Session]: If enter_session is True:
-            A tuple of the current fw and the tf.Session if fw="tf".
+        If `session` is False: The current framework [tf2|tf|tfe|torch] used.
+        If `session` is True: A tuple consisting of the current framework
+        string and the tf1.Session (if fw="tf", otherwise None).
     """
+    global FRAMEWORK_ITERATOR_LAST_RUN_TIMES
```

Comment:
hmm, not sure if global variable will work if tests are run in parallel?
why not pass in a framework_iter_last_run_times dict instead as the container to collect the stats?
if the param is None, we don't have to write the stats.

Reply:
True, this was not a good idea :) Changed.